examples/api/lib/widgets/inherited_model/inherited_model.0.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for InheritedModel
+
+import 'package:flutter/material.dart';
+
+enum Logo { backgroundColor, size }
+
+void main() => runApp(const InheritedModelApp());
+
+class InheritedModelApp extends StatelessWidget {
+  const InheritedModelApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: InheritedModelExample(),
+    );
+  }
+}
+
+class LogoModel extends InheritedModel<Logo> {
+  const LogoModel({
+    super.key,
+    this.backgroundColor,
+    this.logoSize,
+    required super.child,
+  });
+
+  final Color? backgroundColor;
+  final double? logoSize;
+
+  static LogoModel? of(BuildContext context, Logo aspect) {
```

Comment:
There is some danger that you accidentally use this method, pass `aspect: size`, access the background color and then be surprised that you don't rebuild when the background color changes. To avoid this, I would remove the general of method altogether and instead introduce two separate `colorOf` and `sizeOf` methods that only return the relevant value from the model internally using the aspect. (we should also make the `LogoAspect` class private then)